cardano-node/src/Cardano/Node/Run.hs

```diff
@@ -401,6 +418,45 @@ canonDbPath NodeConfiguration{ncDatabaseFile = DbFile dbFp} = do
   createDirectoryIfMissing True fp
   return fp
 
+#ifdef UNIX
+-- | Make sure the VRF private key file is readable only
+-- by the current process owner the node is running under.
+checkVRFFilePermissions :: Maybe FilePath -> ExceptT VRFPrivateKeyFilePermissionError IO ()
+checkVRFFilePermissions mFp =
+  case mFp of
+    Just vrfPrivKey -> do
+      fs <- liftIO $ getFileStatus vrfPrivKey
+      let fm = fileMode fs
+
+      -- Check the the VRF private key file does not give read/write/exec permissions to others.
+      when (hasOtherPermissions fm)
+           (left $ OtherPermissionsExist vrfPrivKey)
+      -- Check the the VRF private key file does not give read/write/exec permissions to any group.
+      when (hasGroupPermissions fm)
+           (left $ GroupPermissionsExist vrfPrivKey)
+      -- Check that the VRF private key file has owner read permissions.
+      unless (hasOwnerPermissions fm)
+             (left $ OwnerDoesNotHaveReadWrite vrfPrivKey)
+
+    Nothing -> right ()
+ where
+
+  hasPermission :: FileMode -> FileMode -> Bool
+  hasPermission fModeA fModeB = fModeA `intersectFileModes` fModeB == fModeB
```

Comment:
Note that this happens to work but only if the fModeB is a single-bit mode. The proper test is to do the intersection and see if the result is non-empty (i.e. not 0).

What's really going on here is that mode a is the word to test, mode b is the bit mask, and by doing a bitwise and we leave 1-bits for all the cases where the word under test has a 1 in a place where the bit mask has a 1. We can check for any 1s by checking if the result is 0 or not.